packages/php-wasm/compile/build-assets/php_wasm.c

```diff
@@ -802,10 +815,38 @@ static void wasm_sapi_register_server_variables(zval *track_vars_array TSRMLS_DC
 	if (value != NULL) {
 		php_register_variable("SCRIPT_NAME", value, track_vars_array TSRMLS_CC);
 		php_register_variable("SCRIPT_FILENAME", value, track_vars_array TSRMLS_CC);
-		php_register_variable("PHP_SELF", value, track_vars_array TSRMLS_CC);
 		php_register_variable("REQUEST_URI", value, track_vars_array TSRMLS_CC);
 	}
 
+	int php_self_set = 0;
+	if (wasm_server_context->document_root != NULL && wasm_server_context->path_translated != NULL)
+	{
+		/**
+		 * PHP_SELF is the script path relative to the document rooth.
+		 *
+		 * For example:
+		 * 
+		 * If the document root is /var/www/html and the requested path is /dir/index.php,
+		 * PHP_SELF will be /dir/index.php.
+		 * 
+		 * If the document root is /var/www/html and the requested path /nice/urls is
+		 * served from /var/www/html/another/directory/script.php, PHP_SELF will be
+		 * /another/directory/script.php
+		 */
+		// Confirm path translated starts with the document root
```

Comment:
we may have two confused comments here. if we want docblock style comments, then we have disconnected it from the code node it describes the the insertion of the line-level comment

we might want to merge these comments or shift them around so that they don't interfere with each other in the docs tooling.

Reply:
added here https://github.com/WordPress/wordpress-playground/pull/280/commits/92b16067071b430725c7672cccbfc4f1dcb95ead